validator/testcases/regex.py

```diff
@@ -450,6 +451,33 @@ def tests(self):
         )
 
 
+@register_generator
+class Gecko50RegexTests(CompatRegexTestHelper):
+    """Regex tests for Firefox 50 updates."""
+
+    VERSION = FX50_DEFINITION
+
+    def tests(self):
+        instances = ('draggesture', 'dragdrop', 'ondraggesture', 'ondragdrop')
+
+        msg = (
+            'The draggesture and dragdrop events are no longer supported. '
+            'You should use the standard equivalents instead: '
+            'dragstart and drop.')
+
+        description = (
+            msg + ' See %s for more information.'
+            % MDN_DOC % 'Web/API/HTML_Drag_and_Drop_API')
+
+        yield self.get_test(
```

Comment:
That's not a unittest but our compat tests with are built around
generators :)


On Wed, Sep 7, 2016, at 11:48 AM, Andreas Wagner wrote:
> In validator/testcases/regex.py[1]:


> > + > +    VERSION = FX50_DEFINITION > + > +    def tests(self): > +
> >   instances = ('draggesture', 'dragdrop', 'ondraggesture',
> >   'ondragdrop') > + > +        msg = ( > +            'The
> >   draggesture and dragdrop events are no longer supported. ' > +
> >   'You should use the standard equivalents instead: ' > +
> >   'dragstart and drop.') > + > +        description = ( > +
> >   msg + ' See %s for more information.' > +            % MDN_DOC %
> >   'Web/API/HTML_Drag_and_Drop_API') > + > +        yield
> >   self.get_test(
>
> No more yield please. Use @pytest.mark.parametrize.
> http://doc.pytest.org/en/latest/parametrize.html#pytest-mark-parametrize
> — You are receiving this because you authored the thread. Reply to
> this email directly, view it on GitHub[2], or mute the thread[3].
>

Links:

  1. https://github.com/mozilla/amo-validator/pull/466#discussion_r77791906
  2. https://github.com/mozilla/amo-validator/pull/466/files/9a22865ee6c3ba4a2fff61fa57e358f8005e4cab#r77791906
  3. https://github.com/notifications/unsubscribe-auth/AAIfGYa5lhxP-Q7g-TZfLLFQTp-8wRE8ks5qnoh2gaJpZM4J2s5A